### ISSUE TITLE
Actor component editor inspection UI 'Bounding box configuration' opens UI elements under a different section

### ISSUE THREAD

**Issue by smurly** [o3de/o3de#9915]
**Describe the bug**
when interacting with the expand control 'Bounding box configuration' the expanded UI elements actually appear below the expand section 'Render options'

**Assets required**
o3de editor.exe 

**Steps to reproduce**
1. Open a new level in editor.exe
2. create a new entity and add component Actor to it
3. click the expand carrot next to 'Bounding box configuration'

**Expected behavior**
elements expand under 'Bounding box configuration'

**Actual behavior**
elements expand under 'Render options'

**Screenshots/Video**
![actor_editor](https://user-images.githubusercontent.com/26234397/171963605-f12ddd3c-8b75-4ab9-b3cf-a0ace495da44.png)

**Found in Branch**
commit 012787e06d497dccbf5de8cad163a539771bfda7 (HEAD -> development

**Comment by amzn-jillich** [comment]
Just double checked in the code of the `EditorActorComponent `and it looks like an issue in the RPE to me. The "Bounds type" and "Expand by" are reflected with a sub-class `ActorComponent::BoundingBoxConfiguration` and then added to the `EditContext `via adding the `DataElement ``m_bboxConfig`, so they should appear directly under their class and not on other locations.

@o3de/sig-content , can you please have a look at this?

**Comment by AMZN-daimini** [comment]
Yeah we were giving it a look during triage and it did not look like it was blatantly incorrect in the reflection side. There have been some changes in handling RPE item ordering recently so maybe it exposed something... We should give it a look from the editor side.
